includes/class-wc-youcanpay-api.php

```diff
@@ -4,247 +4,268 @@
+    {
+        self::$is_test_mode = ('yes' === $test_mode);
+    }
+
+    /**
+     * Get secret key.
+     *
+     * @return string
+     */
+    public static function get_private_key()
+    {
+        if (!self::$private_key) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'], $options['private_key'], $options['sandbox_private_key'])) {
+                self::set_private_key(
+                    'yes' === $options['sandbox_mode'] ? $options['sandbox_private_key'] : $options['private_key']
+                );
+            }
+        }
+
+        return self::$private_key;
+    }
+
+    /**
+     * Get public key.
+     *
+     * @return string
+     */
+    public static function get_public_key()
+    {
+        if (!self::$public_key) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'], $options['public_key'], $options['sandbox_public_key'])) {
+                self::set_public_key(
+                    'yes' === $options['sandbox_mode'] ? $options['sandbox_public_key'] : $options['public_key']
+                );
+            }
+        }
+
+        return self::$public_key;
+    }
+
+    /**
+     * Get public key.
+     *
+     * @return string
+     */
+    public static function is_test_mode()
+    {
+        if ('' === self::$is_test_mode) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'])) {
+                self::set_test_mode($options['sandbox_mode']);
+            }
+        }
+
+        return self::$is_test_mode;
+    }
+
+    /**
+     * Send the request to YouCan Pay's API
+     *
+     * @param WC_Order|WC_Order_Refund $order
+     * @param array $post_data
+     *
+     * @return stdClass|array
+     */
+    public static function request($order, $post_data)
+    {
+        $response = [
+            'id'       => 0,
+            'redirect' => [
+                'url' => get_home_url(),
+            ],
+        ];
+
+        try {
+            if (self::is_test_mode()) {
+                YouCanPay::setIsSandboxMode(true);
+            }
+            $py = YouCanPay::instance()->useKeys(
+                self::get_private_key(),
+                self::get_public_key()
+            );
+
+            $token = $py->token->create(
+                $order->get_id(),
+                (int)$post_data['amount'],
+                $post_data['currency'],
+                self::get_the_user_ip(),
+                $post_data['redirect']['return_url'],
+                $post_data['redirect']['return_url']
+            );
+
+            if (is_wp_error($token) || empty($token)) {
+                WC_YouCanPay_Logger::info('there was a problem connecting to the YouCan Pay API endpoint', [
+                    'order_id' => $order->get_id(),
+                ]);
+
+                throw new WC_YouCanPay_Exception(
```

Comment:
Unhandled exception

Reply:
https://github.com/NextmediaMa/youcan-pay-for-woocommerce/pull/12/commits/8f619601e2f3498862395a06dff350c1928dc177